### PR TITLE
Fixed navbar bug (showing wrong component on clicking back button)

### LINKED ISSUE #109: Navbar links not updating their state with the change in URL
### Description
<write your description>Navbar links are not changing their state in accordance with the change in URL. Refer to the Screenshot attached below , page is in home section while navbar menu is showing Event as active component.
Note : This happens when we press Back button on the browser or phone .

### Screenshots/testimonials
<place your screenshots or testimonials here if applicable>

House of Geeks, IIIT Ranchi - Google Chrome 2020-09-27 22-28-56_Trim (1)

### PR DESCRIPTION
**Pull request by RishavMz**
fixes #109 

Description: Reloading page whenever back button is pressed to make navbar show correct selected component. 

#### Checklist
- [x] I've followed the Contributing guidelines provided in the repository.
- [x] I've made the changes which were demanded in the linked issue.
- [x] I've tested my code on a Chromium based browser.
- [x] I've tested my code on Mozilla Firefox.
- [ ] My code gave a clean console on debugging. (no warnings/errors)

#### Testimonial
Unable to upload video

### CODE REVIEW

**Comment by ankiiitraj** [comment]
you can convert your video to gif and upload it here!

**Comment by RishavMz** [comment]
House of Geeks, IIIT Ranchi - Google Chrome 2020-10-01 18-24-30

**Review by tusharjain0022 — APPROVED** [review]
LGTM ! 💯

**Comment by ankiiitraj** [comment]
I think we should come with another approach to this problem. Reloading the pages takes a lot of time and React works just the opposite.

**Comment by EmperorYP7** [comment]
> I think we should come with another approach to this problem. Reloading the pages takes a lot of time and React works just the opposite.

It works fine for me though.

**Review by EmperorYP7 — APPROVED** [review]
LGTM

**Comment by ankiiitraj** [comment]
Nope this will not work, the page is getting refreshed on every back button click!

File: client/src/components/Navbar/Navbar.js [hunk 1]
```diff
@@ -20,6 +20,11 @@ function Navbar() {
   const [active, setActive] = useState(currenttab);
   const [navOpen, toggleNav] = useState(0);
 
+  // Handling the case for back button press.
+  window.onpopstate=function()
+  {
+    window.location.reload();
```
Comment (anktrj) [inline client/src/components/Navbar/Navbar.js]: instead of reloading, update the state here. That'll do the magic

**Review by ankiiitraj — CHANGES_REQUESTED** [review]

**Comment by EmperorYP7** [comment]
> Nope this will not work, the page is getting refreshed on every back button click!

Woah! I completely missed on that one.. 😮

File: client/src/components/Navbar/Navbar.js (hunk 1 above)
Comment (RishavMz) [reply client/src/components/Navbar/Navbar.js]: I tried updating the state , but for some reason its not working out. 
I made a recent commit in which state is being updated.

**Review by EmperorYP7 — APPROVED** [review]
LGTM. Thanks @RishavMz! :smile:

File: client/src/components/Navbar/Navbar.js [hunk 2]
```diff
@@ -17,9 +18,28 @@ function Navbar() {
...
+
+      if (currenttab[currenttab.length - 1] === "?") {
+        //When search bar is used, preventing bug of nothing being active in navbar
+        currenttab = currenttab.substring(0, currenttab.length - 1); //Removing ? sign when search bar used
+      }
+
+      if (currenttab.length === 0) {
+        //When visiting home section
+        currenttab = "home";
+      }
```
Comment (anktrj) [inline client/src/components/Navbar/Navbar.js:39]: marking this as a temporary change because there is code duplication! 
LGTM for now

**Review by ankiiitraj — APPROVED** [review]

File: client/src/components/Navbar/Navbar.js (hunk 1 above)
Comment (anktrj) [reply client/src/components/Navbar/Navbar.js]: I made the right changes @RishavMz. This is working fine now!

**Review by hariom1625 — APPROVED** [review]
LGTM !! 🙌

**Pull request merged**